sql/src/main/java/io/crate/planner/fetch/FetchPushDown.java

```diff
@@ -36,18 +42,42 @@
 
 public class FetchPushDown {
 
-    private static final LinkedHashMap<Symbol, InputColumn> EMPTY_INPUTS = new LinkedHashMap<>(0);
     private static final FetchRequiredVisitor fetchRequiredVisitor = new FetchRequiredVisitor();
     private static final ScoreReferenceDetector SCORE_REFERENCE_DETECTOR = new ScoreReferenceDetector();
+    private final QuerySpec querySpec;
+    private final DocTableRelation docTableRelation;
+    private Field docIdField;
+    private LinkedHashMap<ReferenceIdent, FetchReference> fetchReferences;
+
+
+    public FetchPushDown(QuerySpec querySpec, DocTableRelation docTableRelation) {
+        this.querySpec = querySpec;
+        this.docTableRelation = docTableRelation;
+    }
+
+    public Field docIdField() {
+        return docIdField;
+    }
+
+    public Collection<Reference> fetchRefs() {
+        if (fetchReferences == null || fetchReferences.isEmpty()){
+            return ImmutableList.of();
+        }
+        return Collections2.transform(fetchReferences.values(), new com.google.common.base.Function<FetchReference, Reference>() {
```

Comment:
would it make sense to make the function instance final static instead of creating a new instance on each fetchRefs call?